integration/bootstrap_test.go

```diff
@@ -0,0 +1,56 @@
+package integration
+
+import (
+	"os"
+	"os/exec"
+	"path"
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+)
+
+const (
+	bootstrapBinaryName = "xds-relay"
+)
+
+func TestBootstrap(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "bootstrap integration tests suite")
+}
+
+var bootstrapTestCases = []TableEntry{
+	{
+		Description: "negative test: negative cache TTL",
+		Parameters: []interface{}{
+			"./integration/testdata/bootstrap_configuration_negative_cache_ttl.yaml",
+			"./integration/testdata/keyer_configuration_complete_tech_spec.yaml",
+			"debug",
+			true,
+			"invalid Cache.Ttl: value must be greater than or equal to 0s",
+		},
+	},
+}
+
+var _ = Describe("Integration tests for bootstrapping xds-relay", func() {
+	DescribeTable("table driven integration tests for bootstrapping xds-relay",
+		func(bootstrapConfigFile string, aggregationRulesFile string, logLevel string, wantErr bool, errorMessage string) {
+			dir, err := os.Getwd()
+			Expect(err).To(BeNil())
+
+			// #nosec G204
+			cmd := exec.Command(path.Join(dir, "bin", bootstrapBinaryName),
```

Comment:
are we starting up the xds-relay binary ? I think we need a way to bootup and validate the files and not really start all the background connections et al.

Reply:
added 'validate' and 'serve' modes. Right now they do the exact same things, except 'validate' doesn't call .Serve().